beets/importer.py

```diff
@@ -572,18 +572,24 @@ def remove_duplicates(self, lib):
                 util.prune_dirs(os.path.dirname(item.path),
                                 lib.directory)
 
-    def set_fields(self):
+    def set_fields(self, lib):
         """Sets the fields given at CLI or configuration to the specified
-        values.
+        values, for both the album and all its items.
         """
-        for field, view in config['import']['set_fields'].items():
-            value = view.get()
-            log.debug(u'Set field {1}={2} for {0}',
-                      displayable_path(self.paths),
-                      field,
-                      value)
-            self.album[field] = value
-        self.album.store()
+        with lib.transaction():
+            items = self.imported_items()
+            for field, view in config['import']['set_fields'].items():
+                value = view.get()
+                log.debug(u'Set field {1}={2} for {0}',
+                          displayable_path(self.paths),
+                          field,
+                          value)
+                self.album[field] = value
```

Comment:
The exact same loop as before, wrapped in a transaction plus amended with changing all items of the album.